src/main/java/hudson/plugins/git/GitStatus.java

```diff
@@ -278,8 +277,21 @@ public void writeBody(PrintWriter w) {
                             }
                             if (!branchFound) continue;
                             urlFound = true;
-
                             if (!(project instanceof AbstractProject && ((AbstractProject) project).isDisabled())) {
+                                //JENKINS-30178 Add default parameters defined in the job
+                                if (project instanceof Job) {
+                                        List<ParameterValue> jobParametersValues = getDefaultParametersValues((Job) project);
+                                        ListIterator<ParameterValue> buildParameterListIterator = buildParameters.listIterator();
+                                        while (buildParameterListIterator.hasNext()) {
+                                            buildParameterListIterator.next();
```

Comment:
What this will do is iterate the `jobParametersValues` array `buildParameters.size()` times... I suspect you have one too many loops so that line https://github.com/jenkinsci/git-plugin/pull/356/files#diff-80bbaceee5cb79587a6a9848e65b9246R333 will execute `buildParameters.size()*buildParameters.size()*jobParametersValues.size()` times... except that as soon as you hit https://github.com/jenkinsci/git-plugin/pull/356/files#diff-80bbaceee5cb79587a6a9848e65b9246R289 you will get a ConcurrentModificationException the next time you  get to https://github.com/jenkinsci/git-plugin/pull/356/files#diff-80bbaceee5cb79587a6a9848e65b9246R285...

Actually this is a full fledged :bug: because of that which is worse than the unnecessary double looping I originally was complaining about